lib/eth_web_socket/server.ex

```diff
@@ -0,0 +1,246 @@
+defmodule EthWebSocket.Server do
+  use GenServer
+  alias EthWebSocket.Client
+  @time_out 60_000
+
+  def start_link(opts \\ []) do
+    GenServer.start_link(__MODULE__, :ok, opts)
+  end
+
+  def init(:ok) do
+    {:ok,
+     %{
+       requests: 0,
+       new_heads: %{request_id: nil, sub: [], subscription_id: nil},
+       new_pending_transactions: %{request_id: nil, sub: [], subscription_id: nil}
+     }}
+  end
+
+  def start_server do
+    {:ok, pid} = start_link()
+    init_ws(pid)
+    pid
+  end
```

Comment:
`start_server` is spawning an `EthWebSocket.Server` process and is sending a message to that process to start an `EthWebSocket.Client`. Is this the intended behaviour?

Reply:
If that's the intended behaviour, then some things could be improved.

1. When I read `start_server`, I immediately assume that an `EthWebSocket.Server` process is being started. And that's true, but it's not the only thing happening, meaning something is being hidden away.
2. From the Wrapeth Client's perspective, will they need to call `EthWebSocket.Server.start_server()`? Or could they add `EthWebSocket.Server` to their applications supervision tree?

Reply:
I tried to make it better in this commit, but I'm kinda struggling to figure out the best way to make everything clear on how it's supposed to work.

Reply:
Try to think like if you wanted to use this Webscokets Wrapeth functionality in another app. What's the best way to integrate it?
Normally, GenServers will be added to the app supervision tree, but in this case, it seems we have just a manual way of doing it, with `start_server/0`